### ISSUE TITLE
Wrong AST node

### ISSUE THREAD

**Issue by R4c00n** [phpmd/phpmd#802]
- PHPMD version: dev-master (cf8677bbd8c291f7f6ed6437f5c941a49ebc2473)
- PHP Version: 7.4
- Installation type: composer
- Operating System / Distribution & Version: MacOS Mojave, Windows10

### Current Behavior
Note: We have dev-master of phpmd and pdepend installed to get PHP7.4 features support

Running phpmd on code like this `$arr[$foo]['bar']['baz'] = 'yes';` results in an undefined method error ` Uncaught Error: Call to undefined method PDepend\Source\AST\ASTArrayIndexExpression::getNode() in path/to/project/vendor/phpmd/phpmd/src/main/php/PHPMD/Rule/AbstractLocalVariable.php:214`

### Expected Behavior
PHPMD command does not fail with a fatal error

### Steps To Reproduce:
1. Install phpmd and pdepend with version `dev-master`
2. Create a file `phpmderror.php` with the following contents:
```php
<?php

class Foo
{
    public function bar(): void
    {
        $foo = 'foo';
        $arr = [];
	$arr[$foo]['bar']['baz'] = 'yes';
    }
}
```
3. Create a file `phpmd.xml` with the following contents:
```xml
<?xml version="1.0"?>
<ruleset name="Demo Ruleset"
         xmlns="http://pmd.sf.net/ruleset/1.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0
                     http://pmd.sf.net/ruleset_xml_schema.xsd"
         xsi:noNamespaceSchemaLocation="
                     http://pmd.sf.net/ruleset_xml_schema.xsd">
    <description>
        PhpMd demo ruleset
    </description>

    <rule ref="rulesets/cleancode.xml">
        <exclude name="StaticAccess" />
    </rule>
    <rule ref="rulesets/codesize.xml" />
    <rule ref="rulesets/unusedcode.xml" />
    <rule ref="rulesets/controversial.xml/Superglobals" />
    <rule ref="rulesets/design.xml" />
</ruleset>
```
4. Run `./vendor/bin/phpmd phpmdbug.php json phpmd.xml`

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by tvbeek** [comment]
@R4c00n I created #805 to fix this. Do you have the possibility to check it?
